src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -443,9 +486,40 @@ internal void InvokeResponse(InvocationResponse invokeResponse)
                 {
                     context.ResultSource.TrySetException(responseEx);
                 }
+                finally
+                {
+                    // Free memory allocated by the host (for input bindings)
+                    if (!_sharedMemoryManager.TryFreeSharedMemoryMapsForInvocation(invokeResponse.InvocationId))
+                    {
+                        _workerChannelLogger.LogWarning($"Cannot free all shared memory resources for invocation: {invokeResponse.InvocationId}");
+                    }
+
+                    // List of shared memory maps that were produced by the worker (for output bindings)
+                    IList<string> outputMaps = GetOutputMaps(invokeResponse.OutputData);
+                    if (outputMaps.Any())
+                    {
+                        // If this invocation was using any shared memory maps produced by the worker, close them to free memory
+                        SendCloseSharedMemoryResourcesForInvocationRequest(outputMaps);
+                    }
+                }
             }
         }
 
+        /// <summary>
+        /// Request to free memory allocated by the worker (for output bindings)
+        /// </summary>
+        /// <param name="outputMaps">List of names of shared memory maps to close from the worker.</param>
+        internal void SendCloseSharedMemoryResourcesForInvocationRequest(IList<string> outputMaps)
+        {
+            CloseSharedMemoryResourcesRequest closeSharedMemoryResourcesRequest = new CloseSharedMemoryResourcesRequest();
```

Comment:
I see this grpc change went in in an earlier change. My thought was that it might have been good to generalize the message to something like InvocationComplete. That way you can send along the info you need for these purposes, and in the future if there is other cleanup we need to do for an invocation, we aren't creating and sending additional messages for each operation we need to do. @pragnagopa ?

Reply:
I'll let @pragnagopa provide input here.

Reply:
That sounds like a good idea. @gohar94 - leave it up to you - whether you want to update proto file now or send a follow up PR after this PR is merged.

Reply:
My vote is to change the proto separately in a subsequent PR and let this one flow. Essentially we just want to rename the message from `CloseSharedMemoryResourcesRequest` to `InvocationComplete`, right?